test/gtest/ternary_tensor_ops.cpp

```diff
@@ -0,0 +1,452 @@
+            }
+
+            auto checkStride = [p = packed](const std::vector<int>& lens,
+                                            const std::vector<int>& strides) {
+                if(p)
+                    return true;
+
+                if(lens.size() > strides.size())
+                    return false;
+
+                // only sparsed case allowed, since all the kernels do not support the last
+                // dimension strides
+                if(strides.back() == 1)
+                {
+                    auto packedStrides =
+                        miopen::TensorDescriptor(miopen_type<T>{}, lens).GetStrides();
+                    return std::equal(packedStrides.rbegin(),
+                                      packedStrides.rend(),
+                                      strides.rbegin(),
+                                      [](int ps, int s) { return s >= ps; });
+                }
+
+                // currently tensor operations do not support non-one stride in the last dimention.
+                return false;
+            };
+
+            if(!checkStride(tensorALens, stride_a))
+                continue;
+            if(!checkStride(tensorBLens, stride_b))
+                continue;
+            if(!checkStride(tensorALens, stride_c))
+                continue;
+
+            for(const auto& alphabeta : alphabetaArr)
+                for(const auto& operation : operationArr)
+                {
+                    TestCase& testCase = testCases.emplace_back();
+
+                    testCase.tensorlens_ac = tensorALens;
+                    testCase.tensorlens_b  = tensorBLens;
+                    testCase.alphabeta     = alphabeta;
+                    testCase.offsets       = final_offsets;
+                    testCase.packed        = packed;
+                    testCase.operation     = operation;
+                    testCase.stride_a      = stride_a;
+                    testCase.stride_b      = stride_b;
+                    testCase.stride_c      = stride_c;
+                }
+        }
+}
+
+template <typename T>
+inline auto GetCases()
+{
+    static std::vector<TestCase> testCases;
+
+    if(!testCases.empty())
+    {
+        return testing::ValuesIn(testCases);
+    }
+
+    for(const auto& tensorALens : tensorALensArr)
+        for(const auto& tensorBLens : tensorBLensArr)
+        {
+            if(!checkTensorsCompatibility(tensorALens, tensorBLens))
+            {
+                continue;
+            }
+
+            AddTestCases<T>(testCases, tensorALens, tensorBLens);
+        }
+
+    return testing::ValuesIn(testCases);
+}
+
+TEST_P(GPU_TensorOps_FP32, TestFloat) {}
+
+TEST_P(GPU_TensorOps_FP16, TestFloat16) {}
+
+TEST_P(GPU_TensorOps_FP64, TestDouble) {}
+
+INSTANTIATE_TEST_SUITE_P(Smoke, GPU_TensorOps_FP32, GetCases<float>());
+INSTANTIATE_TEST_SUITE_P(Smoke, GPU_TensorOps_FP64, GetCases<double>());
+INSTANTIATE_TEST_SUITE_P(Smoke, GPU_TensorOps_FP16, GetCases<half_float::half>());
```

Comment:
It's not `Smoke`, it is `Full`, it has thousands of cases.
Smoke tests must have 1-2 cases and that's it.

I would recommend to change the way how the test cases are generated and do not use ALL the operations for the full tests, since the main point of the full tests with all the different offsets, sizes, etc is to check tensor traversal and broadcasting, while the operation itself it's not important. (as usual, see #3416)

You can have all the operations in the smoke tests, but with limited number of different tensors geometry.